src/galaxy42_utils.cpp

```
@@ -0,0 +1,50 @@
+
+
+#include "galaxy42_utils.hpp"
+#include <boost/asio.hpp>
+
+#include "libs0.hpp"
+
+std::pair<std::string, int> tunserver_utils::parse_ip_string(const std::string &ip_string) {
+	if (ip_string.size()<(1+1)*4) _throw_error(runtime_error(join_string_sep("Invalid (too small) IP size",ip_string.size())));
+	if (ip_string.size()>(4*4+1+4)) _throw_error(runtime_error(join_string_sep("Invalid (too big) IP size",ip_string.size())));
+	/*
+	std::regex pattern("\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}:\\d{1,5}"); // i.e. 127.0.0.1:4562
+	std::smatch result;
+	if (!std::regex_search(ip_string, result, pattern)) { // bad argument
+//		throw std::invalid_argument("bad format of input ip address");
+                throw std::invalid_argument(boost::locale::gettext("L_bad_format_of_input_ip_address"));
+
+	}
+	*/
+	size_t pos = ip_string.find(':');
+	if (pos == string::npos) _throw_error( runtime_error( join_string_sep("Invalid IP format (char ':')", ip_string)));
```

we should throw std::invalid_argument